### PR TITLE
Math magicians: events

### PR DESCRIPTION
**Pull request by musangamfure**
**Hello,** 

In this PR, I have worked on the calculator functionality feature to ensure that All buttons are working properly.

**Here are are what I have achieved:**

- [ ] Create a new directory called logic inside your project
- [ ] Copy the files calculate.js and operate.js into your logic/ directory
- [ ] Install `big.js` using NPM
- [ ] Import calculate.js in your Calculator.js component
- [ ] Implement click event handlers using the math logic from calculate.js and operate.js in my React component

Kindly check and let me know what you think.

**Best regards.**

Happy Coding🎉

### CODE REVIEW

File: src/components/calculator.js [hunk 1]
```diff
@@ -1,44 +1,65 @@
...
-        =
-      </button>
+      <Button name="AC" className="" clickEvent={() => clickEvent('AC')} />
+      <Button name="+/-" className="" clickEvent={() => clickEvent('+/-')} />
+      <Button name="%" className="" clickEvent={() => clickEvent('%')} />
+      <Button
+        name="÷"
+        className="side-color"
+        clickEvent={() => clickEvent('÷')}
+      />
```
Comment (anagudelogu) [inline src/components/calculator.js:31]: - [ ] You did a great job working on this milestone! 💯 However, I see that you are repeating the execution of the function here on your `clickEvent` prop and also in your `Button` component when you call again the content of the prop. In order to follow the DRY (Do not repeat yourself) principle, I suggest that you pass down only the function, in this case, `clickEvent` and let the `Button` component handle the execution logic. This practice is called [lifting state up](https://react.dev/learn/sharing-state-between-components#lifting-state-up-by-example). 

Something like this:


```js
function Calculator() {
  // Change from function to const so the linter doesn't complain
  const clickEvent = (buttonName) => {
    const solution = calculate({ total, next, operation }, buttonName);
    setTotal(solution.total);
    setNext(solution.next);
    setOperation(solution.operation);
  }

  return(
    ...
    <Button name="AC" className="" clickEvent={clickEvent} />
    ...
  )
}
```

Notice that now you're passing the function, and your Button component is in charge of executing it:

```js
function Button({ name, className = 'side-color', clickEvent }) {
  return (
    <button
      type="button"
      className={className}
      // Here the function gets called with the name 'AC' in this example, and the state
      // of the parent component gets updated.
      onClick={() => clickEvent(name)}
    >
      {name}
    </button>
  );
}
```

Please consider applying this change! 🤓 🔧

**Review by anagudelogu — CHANGES_REQUESTED** [review]
# STATUS:  CHANGES REQUIRED  ♻️ 🚧 

Hi @musangamfure, 
I'm Andres.


While you have done a great job working on this milestone 💻 , there are some issues that you still need to work on to go to the next project. ♻️ 

You are almost there, keep up the good work! 🚀 



## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Great UI design. 💻 
✅ Descriptive PR title, summary and commit messages. 💬 

## Required Changes ♻️
_Check the comments under the review._

  - [ ] You did a great job working on this milestone! 🚀 However, to ensure your documentation is professional, please always double-check that all links are working. Currently, your license link is broken due to a typo in its name, this link is looking for a file named `MIT.md` but the actual name is `LICENSE`. Please consider fixing this! 🤓 🔧

image


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/calculator.js (hunk 1 above)
Comment (musangamfure) [reply src/components/calculator.js:31]: Hello, @anagudelogu 

Thank you so much for your suggestion, my brother. but I want to know if it is required or optional

Comment (zainsadaqat) [reply src/components/calculator.js:31]: Hi @musangamfure 
The previous Code Reviewer suggested you some Best Practices and those are optional and you can try to implement that when you've time 👍

**Review by zainsadaqat — APPROVED** [review]
# Pull Request Approved ✅ 🎊 🎉 
## Hello @musangamfure  👋 

### Good job on fixing the issue in the README file requested by the previous Code Reviewer. Your project is complete!  There is nothing else to say other than it's time to merge it.

# Congratulations!!! 🎉 🥳 👏 🙌 🎊 ☑️ 



Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/calculator.js (hunk 1 above)
Comment (musangamfure) [reply src/components/calculator.js:31]: Ooh! Thank you🙏🙏 @zainsadaqat for the clarification. I will implement the suggested changes in the final touches👌

**Comment by musangamfure** [comment]
Hello @zainsadaqat 

Thank you🙏🙏  for approving my PR.

**Pull request merged**